platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,160 @@
+
+    // get the first metadata
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+    it.next();
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+
+    String entityID = testEntry.getKey();
+    EntityInformation testInfo = testEntry.getValue();
+
+    assertThat("Entity ID is set", entityID, is(equalTo(ENTITY_ID)));
+    assertThat(
+        "Signing certificate is set", testInfo.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Encryption certificate is set",
+        testInfo.getEncryptionCertificate(),
+        is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Redirect logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+    assertThat(
+        "Post logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_POST).getUrl(),
+        is(equalTo(POST_LOGOUT_URL)));
+    assertThat(
+        "Redirect assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, 0).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+    assertThat(
+        "Post assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, 1).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+    assertThat(
+        "SOAP assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.SOAP, 2).getUrl(),
+        is(equalTo(SOAP_ACS_URL)));
+  }
+
+  @Test
+  public void parseSecondMetadata() throws Exception {
+
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+
+    // get the second metadata
+    Map.Entry<String, EntityInformation> testEntry = testMap.entrySet().iterator().next();
+
+    String entityID = testEntry.getKey();
+    EntityInformation testInfo = testEntry.getValue();
+
+    assertThat("Entity ID is set", entityID, is(equalTo(ENTITY_ID_2)));
+    assertThat(
+        "Signing certificate is set", testInfo.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Encryption certificate is set",
+        testInfo.getEncryptionCertificate(),
+        is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Redirect logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+    assertThat(
+        "Post logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_POST).getUrl(),
+        is(equalTo(POST_LOGOUT_URL)));
+    assertThat(
+        "Post assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, 0).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+    assertThat(
+        "Artifact assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, 1).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
```

Comment:
❗️ We should add an assertion that the SOAP url is not there.

Reply:
✏️  Also need to improve the failure messages for all of the `assertThat` calls.